src/cider/nrepl/middleware/util/java.clj

```diff
@@ -73,9 +73,19 @@
        (str/replace "$" ".")
        (str ".html")))
   ([class member argtypes]
-   (str (javadoc-url class) "#" member
-        (when argtypes
-          (str "(" (str/join ",%20" argtypes) ")")))))
+   (let [java-version (Integer/parseInt util/java-api-version)
+         old-format   (< java-version 8)
+         open-paren   (if old-format "(" "-")
```

Comment:
Let's just have two string templates - one for old and one for the new format and insert into them whatever's needed. Or just build the new old or new string without actually introducing those locals. They serve no purpose and their names are confusing for the new format (e.g. close-paren is a "-"). I'd rather have two functions that build the old/new format or something like this.